plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -698,7 +699,34 @@ public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, Li
     @Override
     protected Optional<TopNFunction> topNFunction()
     {
-        return Optional.of(TopNFunction.sqlStandard(this::quoted));
+        return Optional.of((query, sortItems, limit) -> {
+            String orderBy = sortItems.stream()
+                    .map(sortItem -> {
+                        String ordering = sortItem.getSortOrder().isAscending() ? "ASC" : "DESC";
+                        String nullsHandling = sortItem.getSortOrder().isNullsFirst() ? "NULLS FIRST" : "NULLS LAST";
+                        String collation = "";
+                        if (isCollatable(sortItem.getColumn())) {
+                            collation = "COLLATE \"C\"";
+                        }
+                        return format("%s %s %s %s", ((Function<String, String>) this::quoted).apply(sortItem.getColumn().getColumnName()), collation, ordering, nullsHandling);
+                    })
+                    .collect(joining(", "));
+
+            return format("%s ORDER BY %s OFFSET 0 ROWS FETCH NEXT %s ROWS ONLY", query, orderBy, limit);
```

Comment:
you can remove `OFFSET 0 ROWS` now (it was needed for SQL Server)
or even switch back to simpler `ORDER BY ... LIMIT ...`.